docs/book/src/tasks/change-machine-template.md

```diff
@@ -0,0 +1,35 @@
+# Changing Infrastructure Machine Templates
+
+Several different components of Cluster API leverage _infrastructure machine templates_, 
+including `KubeadmControlPlane`, `MachineDeployment`, and `MachineSet`. These 
+`MachineTemplate` resources are **immutable** and are not intended to be modified 
+directly.
+
+The correct process for modifying an infrastructure machine template is as follows:
+
+1. Duplicate an existing template.
+    Users can use `kubectl get <MachineTemplateType> <name> -o yaml > file.yaml` 
+    to retrieve a template configuration from a running cluster to serve as a starting 
+    point.
+2. Update the desired fields.
+    Fields that might need to be modified could include the SSH key, the AWS instance 
+    type, or the Azure machine type). Refer to the provider-specific documentation 
+    for more details on the specific fields that each provider requires or accepts.
+3. Give the newly-modified template a new name by modifying the `metadata.name` field.
+4. Create the new infrastructure machine template on the API server using `kubectl`.
+
+Once the new infrastructure machine template has been persisted, users may modify 
+the object that was referencing the infrastructure machine template. For example, 
+to modify the infrastructure machine template for the `KubeadmControlPlane` object, 
+users would modify the `spec.infrastructureTemplate.name` field. For a `MachineDeployment` 
+or `MachineSet`, users would need to modify the `spec.template.spec.infrastructureRef.name` 
+field. In all cases, the `name` field should be update to point to the newly-modified 
+infrastructure machine template. This will trigger a rolling update. (This same process 
+is described in the documentation for [upgrading the underlying machine image for 
+KubeadmControlPlane](./kubeadm-control-plane.html#how-to-upgrade-the-underlying-machine-image).)
```

Comment:
I don't think CI likes this link

Reply:
The link worked when I ran `mdbook` locally, but I've simplified the link in af55bb4 and hopefully this will work now.